src/ui/components/ExplorerHome/ExplorerHome.js

```diff
@@ -225,50 +325,108 @@ export const ExplorerHome = ({initialView}: ExplorerHomeProps): ReactNode => {
                     <b>Participant</b>
                   </TableCell>
                   <TableCell>
-                    <b>Cred</b>
+                    <TableSortLabel
+                      active={tsParticipants.sortName === CRED_SORT.name}
+                      direction={
+                        tsParticipants.sortName === CRED_SORT.name
+                          ? tsParticipants.sortOrder
+                          : DEFAULT_SORT
+                      }
+                      onClick={() =>
+                        tsParticipants.setSortFn(CRED_SORT.name, CRED_SORT.fn)
+                      }
+                    >
+                      <b>{CRED_SORT.name.description}</b>
+                    </TableSortLabel>
                   </TableCell>
                   <TableCell>
-                    <b>Grain</b>
+                    <TableSortLabel
+                      active={tsParticipants.sortName === GRAIN_SORT.name}
+                      direction={
+                        tsParticipants.sortName === GRAIN_SORT.name
+                          ? tsParticipants.sortOrder
+                          : DEFAULT_SORT
+                      }
+                      onClick={() =>
+                        tsParticipants.setSortFn(GRAIN_SORT.name, GRAIN_SORT.fn)
+                      }
+                    >
+                      <b>{currencyName}</b>
+                    </TableSortLabel>
                   </TableCell>
                   <TableCell>
                     <b>Contributions Chart (ALL TIME)</b>
                   </TableCell>
                 </TableRow>
               </TableHead>
               <TableBody>
-                {rows.map((row) => (
-                  <TableRow key={row.username}>
-                    <TableCell component="th" scope="row">
-                      {row.username}
-                    </TableCell>
-                    <TableCell>{row.cred}</TableCell>
-                    <TableCell>{row.grain}</TableCell>
-                    <TableCell align="right">
-                      <CredTimeline data={row.chart} />
+                {tsParticipants.currentPage.length > 0 ? (
+                  tsParticipants.currentPage.map((row) => (
+                    <TableRow key={row.identity.name}>
+                      <TableCell component="th" scope="row">
+                        {row.identity.name}
+                      </TableCell>
+                      <TableCell>{Math.round(row.cred)}</TableCell>
+                      <TableCell>
+                        {Number(row.grainEarned).toFixed(2) + currencySuffix}
```

Comment:
You should use the `format` function provided in the `core/ledger/grain.js` module instead. (this is because grain is a BigInt, so I believe casting to number can fail for large values that may be stored)

Reply:
Done, thank you. I completely forgot about the special handling of grain calculations.